app/flatpak-builtins-ls-remote.c

```diff
@@ -119,80 +124,90 @@ flatpak_builtin_ls_remote (int argc, char **argv, GCancellable *cancellable, GEr
+              g_autoptr(GVariant) deploy_data = flatpak_dir_get_deploy_data (dir, ref, cancellable, NULL);
+
+              if (deploy_data == NULL)
+                continue;
+
+              if (g_strcmp0 (flatpak_deploy_data_get_commit (deploy_data), checksum) == 0)
+                continue;
+            }
 
-          if (g_strcmp0 (flatpak_deploy_data_get_commit (deploy_data), checksum) == 0)
+          if (arches != NULL && !g_strv_contains (arches, parts[2]))
             continue;
-        }
 
-      if (arches != NULL && !g_strv_contains (arches, parts[2]))
-        continue;
+          if (strcmp (parts[0], "runtime") == 0 && !opt_runtime)
+            continue;
 
-      if (strcmp (parts[0], "runtime") == 0 && !opt_runtime)
-        continue;
+          if (strcmp (parts[0], "app") == 0 && !opt_app)
+            continue;
 
-      if (strcmp (parts[0], "app") == 0 && !opt_app)
-        continue;
+          if (!opt_show_details)
+            name = parts[1];
+          else
+            name = ref;
 
-      if (!opt_show_details)
-        name = parts[1];
-      else
-        name = ref;
+          if (!opt_all &&
+              strcmp (parts[0], "runtime") == 0 &&
+              flatpak_id_has_subref_suffix (parts[1]))
+            {
+              g_autofree char *prefix_partial_ref = NULL;
+              char *last_dot = strrchr (parts[1], '.');
 
-      if (!opt_all &&
-          strcmp (parts[0], "runtime") == 0 &&
-          flatpak_id_has_subref_suffix (parts[1]))
-        {
-          g_autofree char *prefix_partial_ref = NULL;
-          char *last_dot = strrchr (parts[1], '.');
+              *last_dot = 0;
+              prefix_partial_ref = g_strconcat (parts[1], "/", parts[2], "/", parts[3], NULL);
+              *last_dot = '.';
 
-          *last_dot = 0;
-          prefix_partial_ref = g_strconcat (parts[1], "/", parts[2], "/", parts[3], NULL);
-          *last_dot = '.';
+              if (g_hash_table_lookup (pref_hash, prefix_partial_ref))
+                continue;
+            }
 
-          if (g_hash_table_lookup (pref_hash, prefix_partial_ref))
-            continue;
-        }
+          if (!opt_all && opt_arch == NULL &&
+              /* Hide non-primary arches if the primary arch exists */
+              strcmp (arches[0], parts[2]) != 0)
+            {
+              g_autofree char *alt_arch_ref = g_strconcat (parts[0], "/", parts[1], "/", arches[0], "/", parts[3], NULL);
+              if (g_hash_table_lookup (refs, alt_arch_ref))
+                continue;
+            }
 
-      if (!opt_all && opt_arch == NULL &&
-          /* Hide non-primary arches if the primary arch exists */
-          strcmp (arches[0], parts[2]) != 0)
-        {
-          g_autofree char *alt_arch_ref = g_strconcat (parts[0], "/", parts[1], "/", arches[0], "/", parts[3], NULL);
-          if (g_hash_table_lookup (refs, alt_arch_ref))
-            continue;
+          if (g_hash_table_lookup (names, name) == NULL)
```

Comment:
This will cause problems if the same ref are in multiple remotes. For instance, the gnome runtimes are both in the gnome repo and the flathub repo.